src/client/views/book/bookEditor.ml

```diff
@@ -44,6 +44,30 @@ let insert t slug i =
 let add t slug =
   insert t slug t.count
 
+let prefill t slug =
+  let%lwt cur = Book.get slug in
+  let%lwt title = Book.title cur in
+  let%lwt date = Book.date cur in
+  let%lwt contents = Book.contents cur in
+  let contents = List.filter_map
+      (function
+          Book.Version _ | Book.InlineSet _ -> None
+        | Book.Set (set, _) -> Some set)
+      contents in
+  let%lwt contents = Lwt_list.map_p
+      (fun s -> let%lwt slug = Set.slug s in
+        Lwt.return (Some (slug, s)))
+      contents in
```

Comment:
Not a big fan of this formatting but that's probably why we should introduce a formatter: so that we don't have to like it or not; there'd be only one formatting and that's it.